web/src/components/AssertionForm/AssertionFormCheck.tsx

```diff
@@ -0,0 +1,111 @@
+import {PlusOutlined} from '@ant-design/icons';
+import {Form, FormInstance, Input, Select} from 'antd';
+import {FormListFieldData} from 'antd/lib/form/FormList';
+import {capitalize} from 'lodash';
+import {CompareOperator} from '../../constants/Operator.constants';
+import {TraceTestAttributes} from '../../constants/SpanAttribute.constants';
+import CreateAssertionModalAnalyticsService from '../../services/Analytics/CreateAssertionModalAnalytics.service';
+import OperatorService from '../../services/Operator.service';
+import {TAssertion} from '../../types/Assertion.types';
+import {IValues} from './AssertionForm';
+import * as S from './AssertionForm.styled';
+import {DurationFields} from './DurationFields';
+
+const operatorList = Object.values(CompareOperator).map(value => ({
+  value: OperatorService.getOperatorSymbol(value),
+  label: capitalize(OperatorService.getOperatorName(value)),
+}));
+
+const getIsValid = ({attribute, comparator, expected}: TAssertion): boolean =>
+  Boolean(attribute && comparator && expected);
+
+interface IProps {
+  add: any;
+  remove: any;
+  form: FormInstance<IValues>;
+  field: Pick<FormListFieldData, never>;
+  name: number;
+  attributeOptionList: JSX.Element[];
+  index: number;
+  length: number;
+  tAssertions: TAssertion[];
+}
+
+export const AssertionFormCheck = ({
+  attributeOptionList,
+  field,
+  index,
+  length,
+  name,
+  tAssertions,
+  form,
+  add,
+  remove,
+}: IProps) => {
+  const assertion = tAssertions?.[index];
+  return (
+    <S.Check>
+      <Form.Item
+        {...field}
+        name={[name, 'attribute']}
+        style={{margin: 0}}
+        rules={[{required: true, message: 'Attribute is required'}]}
+        data-cy="assertion-check-attribute"
+        id="assertion-check-attribute"
+      >
+        <S.Select style={{margin: 0}} placeholder="Attribute" showSearch>
+          {attributeOptionList}
+        </S.Select>
+      </Form.Item>
+      <Form.Item
+        {...field}
+        style={{margin: 0}}
+        name={[name, 'comparator']}
+        rules={[{required: true, message: 'Operator is required'}]}
+        data-cy="assertion-check-operator"
+      >
+        <S.Select style={{margin: 0}} placeholder="Assertion Type">
+          {operatorList.map(({value, label}) => (
+            <Select.Option key={value} value={value}>
+              {label}
+            </Select.Option>
+          ))}
+        </S.Select>
+      </Form.Item>
+      {assertion?.attribute === TraceTestAttributes.TRACETEST_SPAN_DURATION ? (
```

Comment:
We should be checking the attributes using a regex that will tell us if we need to display this extra input or not like:
```
^([0-9]+(\\.[0-9]+)?)(ns|μs|ms|s|m|h)$
```

Instead of having to match for a specific field.

Reply:
i think doing this might the regex might fail depending on what the users types for value

Reply:
We should be able to lock the value input for only numbers in case we find the value matches a duration. I want to avoid having specific custom logic just for one attribute, does it make sense?